src/Core/Services/MetadataProviders/ISqlMetadataProvider.cs

```diff
@@ -123,6 +123,16 @@ public interface ISqlMetadataProvider
         /// <returns></returns>
         public IReadOnlyDictionary<string, DatabaseObject> GetEntityNamesAndDbObjects();
 
+        /// <summary>
+        /// Given entity name, gets the entity to column mappings if present.
+        /// </summary>
+        public bool TryGetEntityToFieldMappings(string entityName, [NotNullWhen(true)] out IReadOnlyDictionary<string, string>? mappings);
+
+        /// <summary>
+        /// Given entity name, gets the column to entity mappings if present.
```

Comment:
nit: the description/function name should be fixed. This is getting the mapping from backing to exposed field names for the given entity. Neither the function name nor the description is suggesting this. Not your doing, it was there previously - but since you are exposing this in the interface now, I think we should fix the naming of both the functions and their descriptions. 
The name FieldToEntityMappings implies - we are retrieving a dictionary that maps a field name to its entity - but thats not what this function is doing

Reply:
Fixed this to new name.